watertap/unit_models/ion_exchange_0D.py

```diff
@@ -141,85 +246,92 @@ class IonExchangeODData(InitializationMixin, UnitModelBlockData):
         ),
     )
 
+    CONFIG.declare(
+        "isotherm",
+        ConfigValue(
+            default=IsothermType.langmuir,
+            domain=In(IsothermType),
+            description="Designates the isotherm type to use for equilibrium calculations",
+        ),
+    )
+
     def build(self):
         super().build()
 
-        ### REFERENCES ###
-        # LeVan, M. D., Carta, G., & Yon, C. M. (2019).
-        # Section 16: Adsorption and Ion Exchange.
-        # Perry's Chemical Engineers' Handbook, 9th Edition.
-
-        # Crittenden, J. C., Trussell, R. R., Hand, D. W., Howe, K. J., & Tchobanoglous, G. (2012).
-        # Chapter 16: Ion Exchange.
-        # MWH's Water Treatment (pp. 1263-1334): John Wiley & Sons, Inc.
-
-        # DOWEX Ion Exchange Resins Water Conditioning Manual
-        # https://www.lenntech.com/Data-sheets/Dowex-Ion-Exchange-Resins-Water-Conditioning-Manual-L.pdf
-
-        # Inamuddin, & Luqman, M. (2012).
-        # Ion Exchange Technology I: Theory and Materials.
-
-        # Michaud, C.F. (2013)
-        # Hydrodynamic Design, Part 8: Flow Through Ion Exchange Beds
-        # Water Conditioning & Purification Magazine (WC&P)
-        # https://wcponline.com/2013/08/06/hydrodynamic-design-part-8-flow-ion-exchange-beds/
-
-        ion_set = self.config.property_package.ion_set
+        comps = self.config.property_package.component_list
         target_ion = self.config.target_ion
+
         self.target_ion_set = Set(
             initialize=[target_ion]
         )  # create set for future development of multi-component model
+        inerts = comps - self.target_ion_set
```

Comment:
Although it is future proofing, your domain in line 226 is still a string. So technically this `target_ion_set` is always initialized to a list of size 1 that contains a string. You could convert over line 226 to a list, but it is completely unnecessary until the model supports multicomponent anyway.

Reply:
Yes right now you could use a list but it would coerce the list to a str anyway... but using a list in target_ion config would perpetuate other errors anyway so I didn't change it. I also dug into this after I had finished prepping the PR, but if you want this change to get it merged I will follow what you have in GAC.

Reply:
No, I think consistency isn't relevant until multicomponent adsorption is actually supported.